cmd/demo-client/main.go

```diff
@@ -159,3 +206,53 @@ func Watch(c *cli.Context, client client.Client) error {
 	}
 	return nil
 }
+
+func newPrometheusBridge(address string, gateway string, pushIntervalSec int64) prometheus.Registerer {
```

Comment:
might be able to simplify this structure (this object takes a registry and exposes it either at an http location or by active pushing) by taking in a registry to expose.

Reply:
Why is this necessary? This is a binary-private method whose only job is to create the registry out of command line arguments. What is the point of "generalizing" it to take a Registry, when I will have to then move the Registry creation code in some other function. There is only one use for this method, that's why it is private.

Reply:
I'm trying to push towards have a consistent pattern across the binaries.

the pattern on cli / relay are patterned to only expose rather than internally manage a registry. This one case is inverting that pattern, and I don't see why. It can instead directly take the `metrics.ClientMetrics` registry that follows the pattern (see #437), and directly either serve or push it. the duplicate structure managing a separate registry for the binary isn't needed.

Reply:
This sounds reasonable, but it is also out of scope for this PR: The effort to make binaries similar (as well as to fold the client into the drand-cli command) are largish separate efforts. This PR is supposed to be tested in one hour in order to derisk it by verifying its functionality. I don't think we should keep adding goals to it in light of future objectives. (Go development is incremental after all, no?)

Also note what is going on here: This comment is justified by an entirely new PR (437) that was just introduced. This is a prime example of scope creep. I am not against fixing this, but PR's shouldn't keep on being extended longer and longer as they are being reviewed. This is really bad practice.